tests/fe/cuthermite_test.cc

```diff
@@ -0,0 +1,1292 @@
+                }
+                else
+                    print_run(std::cout);
+            }
+
+    void 
+    WaveSolver::plot_solution(const std::string &run_name)
+    {
+        this->data_out.set_cell_selection(
+            [this](const typename Triangulation<DIM>::cell_iterator &cell) 
+                {
+                    return cell->is_active() &&
+                        this->mesh_classifier.location_to_level_set(cell) !=
+                        NonMatching::LocationToLevelSet::outside;
+                }
+            );
+        
+        this->data_out.build_patches(
+            this->solution_mapping_collection[0],
+            this->fe_degree,
+            DataOut<DIM>::CurvedCellRegion::curved_inner_cells);
+
+        std::ofstream solution_plots(
+            std::string("wave_equation_") + run_name + std::string(".vtu"));
+        this->data_out.write_vtu(solution_plots);
+        solution_plots.close();
+    }
+
+    void
+    WaveSolver::clean_memory()
+    {
+        this->grid.clear();
+
+        this->solution_mapping_collection = hp::MappingCollection<DIM>();
+        this->solution_fe_collection = hp::FECollection<DIM>();
+        this->solution_quadrature = hp::QCollection<DIM>();
+        this->solution_quadrature_1d = hp::QCollection<1>();
+        
+        delete this->nonmatching_fe_values;
+    }
+
+    void
+    WaveSolver::run(
+        const double start_time,
+        const double end_time,
+        const unsigned int refinement, 
+        const bool plot_sol)
+    {
+        Assert(
+            start_time < end_time,
+            ExcMessage("ERROR: Simulation must have a finite duration."));
+        Assert(
+            (this->fe_degree % 2 == 1), 
+            ExcMessage("ERROR: Provided finite element degree for Hermite "
+                        "interpolation polynomials should be an odd number."));
+
+        this->analytical_solution.set_current_time(start_time);
+
+        make_grid(refinement);
+        setup_levelset();
+
+        distribute_dofs();
+        initialise_quadratures();
+        initialise_matrices();
+        assembly_preprocessing();
+        assemble_system();
+        prepare_system_matrices();
+
+        std::vector<double> l2_errors;
+
+        std::ostringstream name_frag;
+        name_frag << "solution_p_" << this->fe_degree << "_h_" << dx;
+        std::string name_prefix = name_frag.str();
+        for (auto & s : name_prefix)
+            if (s == '.')
+                s = '_';
+
+        solve_in_time(end_time, l2_errors, plot_sol, name_prefix);
+        
+        print_errors(l2_errors, this->fe_degree, true);
+        if (plot_sol)
+            plot_solution(name_prefix);
+
+        clean_memory();
+    }
+} // CutHermiteWaveCircle
+
+
+
+int main() 
+{
+    CutHermiteWaveCircle::WaveSystemValues<DIM> system_parameters;
+
+    for (unsigned int j = 5; j < 6; j += 2)
+    {
+        CutHermiteWaveCircle::WaveSolver system(system_parameters, j);
```

Comment:
The test seems to be quite large. How long does the test take? Do you really want to all of this or just a subset?